app/ImportJob.php

```diff
@@ -526,10 +526,31 @@ public function importEmail(\App\Contact $contact): void
     public function importTel(\App\Contact $contact): void
     {
         if (! is_null($this->formatValue($this->currentEntry->TEL))) {
+            $tel =  (string) $this->currentEntry->TEL;
+
+            if (! is_null($this->formatValue($this->currentEntry->ADR))) {
```

Comment:
- This whole new section should be split, otherwise it'll be very hard to unit test this method, considering the number of `if` involved.
- Ideally we should not have more than one `if` - here we have 3 nested ifs which is not good

You need to refactor this:
* create a new method in an helper (in `LocaleHelper` for instance), something like `getCountryForVCard` or something, that will return the country ISO based on the vcard you pass as parameter
* then create a new method in an helper (in `LocaleHelper` for instance), called `formatTelephoneNumber` or something, that will take both the tel number and the iso country and returns the formatted phone number (or null).
* Then, the entire new code becomes (pseudo code and might contain errors):

```php
$tel = $this->currentEntry->TEL;

if (! is_null($this->formatValue($tel)) {
    $tel = (string) $tel;

    $isoCode = LocaleHelper::getCountryForVCard($this->currentEntry);

    $tel = LocaleHelper::formatTelephoneNumber($tel, $isoCode);

   // and that's it, you have your phone number 
}
```

Then, write the unit tests for the new methods inside the Helpers, and that's it.

The idea is that a method should ideally do just one thing, not multiple things as you've done, and it will be way easier to check if something is wrong.

Reply:
@djaiss Thanks for the reminding, no prob!